libvast/src/system/importer.cpp

```diff
@@ -38,6 +38,79 @@
 
 namespace vast::system {
 
+namespace {
+
+class driver : public importer_state::driver_base {
+public:
+  using super = importer_state::driver_base;
+
+  using pointer = importer_actor*;
+
+  driver(importer_state::downstream_manager& out, pointer self)
+    : super(out), self_(self) {
+    // nop
+  }
+
+  void process(caf::downstream<output_type>& out,
+               std::vector<input_type>& xs) override {
+    VAST_TRACE(VAST_ARG(xs));
+    auto& st = self_->state;
+    uint64_t events = 0;
+    auto t = timer::start(st.measurement_);
+    for (auto& x : xs) {
+      VAST_ASSERT(x.rows() <= static_cast<size_t>(st.available_ids()));
+      auto rows = x.rows();
+      events += rows;
+      x.offset(st.next_id(rows));
+      out.push(std::move(x));
+    }
+    t.stop(events);
+  }
+
+  void finalize(const error& err) override {
+    VAST_DEBUG(self_, "stopped with message:", render(err));
+  }
+
+  pointer self() const {
+    return self_;
+  }
+
+private:
+  pointer self_;
+};
+
+class manager : public caf::detail::stream_stage_impl<driver> {
+public:
+  using super = caf::detail::stream_stage_impl<driver>;
+
+  manager(importer_actor* self) : caf::stream_manager(self), super(self, self) {
```

Comment:
Add a comment explaining the arguments to `super`. Don't init the `stream_manager`.